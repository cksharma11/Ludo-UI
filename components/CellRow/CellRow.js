/* eslint-disable no-plusplus */
import React from 'react';
import PropTypes from 'prop-types';
import Cell from '../Cell/Cell';
import CellRowStyles from './CellRow.style';

const CellRow = ({
  color,
  specialCells,
  className,
  containerClass,
  cellId
}) => {
  let cellIdentifire = cellId - 1;

  const getClassName = (id) => {
    return 'cell '.concat(specialCells.includes(id) ? color : '');
  };

  return (
    <div className={containerClass}>
      {new Array(6).fill('').map(() => {
        return (
          <div className={className}>
            <Cell
              id={++cellIdentifire}
              className={getClassName(cellIdentifire)}
            />
            <Cell
              id={++cellIdentifire}
              className={getClassName(cellIdentifire)}
            />
            <Cell
              id={++cellIdentifire}
              className={getClassName(cellIdentifire)}
            />
          </div>
        );
      })}
      <style jsx>{CellRowStyles}</style>
    </div>
  );
};

CellRow.defaultProps = {
  color: '',
  specialCells: [],
  className: '',
  containerClass: '',
  cellId: 0
};

CellRow.propTypes = {
  color: PropTypes.string,
  specialCells: PropTypes.array,
  className: PropTypes.string,
  containerClass: PropTypes.string,
  cellId: PropTypes.number
};

export default CellRow;
