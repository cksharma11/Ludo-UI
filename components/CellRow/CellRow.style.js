import css from 'styled-jsx/css';

const CellRowStyles = css`
  .cell_row {
    display: flex;
  }

  .cell_column {
    display: flex;
    flex-direction: column;
  }

  .cell_column_container {
    display: flex;
  }
`;

export default CellRowStyles;
