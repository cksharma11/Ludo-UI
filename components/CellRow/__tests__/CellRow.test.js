import React from 'react';
import { shallow } from 'enzyme';
import CellRow from '../CellRow';

describe('CellRow', () => {
  it('should render properly', () => {
    const wrapper = shallow(<CellRow />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render properly with special id', () => {
    const wrapper = shallow(<CellRow specialCells={[1]} id={1} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
