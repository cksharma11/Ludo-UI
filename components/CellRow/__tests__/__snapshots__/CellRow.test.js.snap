// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellRow should render properly 1`] = `
"<div className=\\"jsx-2616575202 \\">
  <div className=\\"jsx-2616575202 \\">
    <Cell id={19} className=\\"cell \\" gameData={{...}} playerId={-1} />
    <Cell id={20} className=\\"cell \\" gameData={{...}} playerId={-1} />
    <Cell id={21} className=\\"cell \\" gameData={{...}} playerId={-1} />
  </div>
  <div className=\\"jsx-2616575202 \\">
    <Cell id={22} className=\\"cell  red safe-cell\\" gameData={{...}} playerId={-1} />
    <Cell id={23} className=\\"cell  red\\" gameData={{...}} playerId={-1} />
    <Cell id={24} className=\\"cell \\" gameData={{...}} playerId={-1} />
  </div>
  <div className=\\"jsx-2616575202 \\">
    <Cell id={25} className=\\"cell \\" gameData={{...}} playerId={-1} />
    <Cell id={26} className=\\"cell  red\\" gameData={{...}} playerId={-1} />
    <Cell id={27} className=\\"cell  safe-cell\\" gameData={{...}} playerId={-1} />
  </div>
  <div className=\\"jsx-2616575202 \\">
    <Cell id={28} className=\\"cell \\" gameData={{...}} playerId={-1} />
    <Cell id={29} className=\\"cell  red\\" gameData={{...}} playerId={-1} />
    <Cell id={30} className=\\"cell \\" gameData={{...}} playerId={-1} />
  </div>
  <div className=\\"jsx-2616575202 \\">
    <Cell id={31} className=\\"cell \\" gameData={{...}} playerId={-1} />
    <Cell id={32} className=\\"cell  red\\" gameData={{...}} playerId={-1} />
    <Cell id={33} className=\\"cell \\" gameData={{...}} playerId={-1} />
  </div>
  <div className=\\"jsx-2616575202 \\">
    <Cell id={34} className=\\"cell \\" gameData={{...}} playerId={-1} />
    <Cell id={35} className=\\"cell  red\\" gameData={{...}} playerId={-1} />
    <Cell id={36} className=\\"cell \\" gameData={{...}} playerId={-1} />
  </div>
  <JSXStyle id=\\"2616575202\\">
    .
    c
    e
    l
    l
    _
    r
    o
    w
    .
    j
    s
    x
    -
    2
    6
    1
    6
    5
    7
    5
    2
    0
    2
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    }
    .
    c
    e
    l
    l
    _
    c
    o
    l
    u
    m
    n
    .
    j
    s
    x
    -
    2
    6
    1
    6
    5
    7
    5
    2
    0
    2
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    }
    .
    c
    e
    l
    l
    _
    c
    o
    l
    u
    m
    n
    _
    c
    o
    n
    t
    a
    i
    n
    e
    r
    .
    j
    s
    x
    -
    2
    6
    1
    6
    5
    7
    5
    2
    0
    2
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    }
  </JSXStyle>
</div>"
`;

exports[`CellRow should render properly with special id 1`] = `
"<div className=\\"jsx-2616575202 \\">
  <div className=\\"jsx-2616575202 \\">
    <Cell id={19} className=\\"cell \\" gameData={{...}} playerId={-1} />
    <Cell id={20} className=\\"cell \\" gameData={{...}} playerId={-1} />
    <Cell id={21} className=\\"cell \\" gameData={{...}} playerId={-1} />
  </div>
  <div className=\\"jsx-2616575202 \\">
    <Cell id={22} className=\\"cell  red safe-cell\\" gameData={{...}} playerId={-1} />
    <Cell id={23} className=\\"cell  red\\" gameData={{...}} playerId={-1} />
    <Cell id={24} className=\\"cell \\" gameData={{...}} playerId={-1} />
  </div>
  <div className=\\"jsx-2616575202 \\">
    <Cell id={25} className=\\"cell \\" gameData={{...}} playerId={-1} />
    <Cell id={26} className=\\"cell  red\\" gameData={{...}} playerId={-1} />
    <Cell id={27} className=\\"cell  safe-cell\\" gameData={{...}} playerId={-1} />
  </div>
  <div className=\\"jsx-2616575202 \\">
    <Cell id={28} className=\\"cell \\" gameData={{...}} playerId={-1} />
    <Cell id={29} className=\\"cell  red\\" gameData={{...}} playerId={-1} />
    <Cell id={30} className=\\"cell \\" gameData={{...}} playerId={-1} />
  </div>
  <div className=\\"jsx-2616575202 \\">
    <Cell id={31} className=\\"cell \\" gameData={{...}} playerId={-1} />
    <Cell id={32} className=\\"cell  red\\" gameData={{...}} playerId={-1} />
    <Cell id={33} className=\\"cell \\" gameData={{...}} playerId={-1} />
  </div>
  <div className=\\"jsx-2616575202 \\">
    <Cell id={34} className=\\"cell \\" gameData={{...}} playerId={-1} />
    <Cell id={35} className=\\"cell  red\\" gameData={{...}} playerId={-1} />
    <Cell id={36} className=\\"cell \\" gameData={{...}} playerId={-1} />
  </div>
  <JSXStyle id=\\"2616575202\\">
    .
    c
    e
    l
    l
    _
    r
    o
    w
    .
    j
    s
    x
    -
    2
    6
    1
    6
    5
    7
    5
    2
    0
    2
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    }
    .
    c
    e
    l
    l
    _
    c
    o
    l
    u
    m
    n
    .
    j
    s
    x
    -
    2
    6
    1
    6
    5
    7
    5
    2
    0
    2
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    }
    .
    c
    e
    l
    l
    _
    c
    o
    l
    u
    m
    n
    _
    c
    o
    n
    t
    a
    i
    n
    e
    r
    .
    j
    s
    x
    -
    2
    6
    1
    6
    5
    7
    5
    2
    0
    2
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    }
  </JSXStyle>
</div>"
`;
